iphone/Maps/Bookmarks/Categories/Sharing/Tags/TagsLayout.swift

```diff
@@ -0,0 +1,109 @@
+final class TagsLayout: UICollectionViewLayout {
+  enum Element: String {
+    case header
+    case cell
+  }
+
+  private var cache = [Element : [IndexPath : UICollectionViewLayoutAttributes]]()
+  fileprivate var contentHeight: CGFloat = 0
+
+  fileprivate var contentWidth: CGFloat {
+    guard let collectionView = collectionView else {
+      return 0
+    }
+    
+    let insets = collectionView.contentInset
+    return collectionView.bounds.width - (insets.left + insets.right)
+  }
+
+  override var collectionViewContentSize: CGSize {
+    return CGSize(width: contentWidth, height: contentHeight)
+  }
+  
+  let itemHeight: CGFloat = 50
+  let itemWidth: CGFloat = 50
+  let lineSpacing: CGFloat = 10
+  let elementsSpacing: CGFloat = 10
```

Comment:
I'd better have public properties for these layout params to make the layout more configurable and flexible